src/app/use-cases/request-signature.ts

```diff
@@ -71,3 +88,32 @@ export const makeRequestSignature =
       ),
       TE.chainFirst(addSignatureRequest)
     );
+
+/*
+ * Update status of signature request only if new status is READY
+ */
+export const updateStatusRequestSignature =
+  (
+    upsertSignatureRequest: UpsertSignatureRequest,
+    getSignatureRequest: GetSignatureRequest
+  ) =>
+  (
+    payload: RequestSignatureStatusPayload
+  ): TE.TaskEither<Error, SignatureRequest> =>
+    pipe(
+      payload.signatureRequestStatus,
+      validate(t.literal("READY"), "Only READY status is allowed!"),
+      TE.fromEither,
+      TE.chain((_) =>
+        getSignatureRequest(payload.signatureRequestId, payload.subscriptionId)
+      ),
+      TE.chainEitherKW(
+        flow(
+          E.fromOption(
+            () => new EntityNotFoundError("Error getting the Signature Request")
+          ),
+          E.chainW((request) => pipe("MARK_AS_READY", nextStatus(request)))
+        )
+      ),
+      TE.chain(upsertSignatureRequest)
```

Comment:
```suggestion
      TE.chain(() =>
        getSignatureRequest(payload.signatureRequestId, payload.subscriptionId)
      ),
      TE.chainW(
        TE.fromOption(
          () => new EntityNotFoundError("Error getting the Signature Request")
        )
      ),
      TE.chainEitherKW((request) => pipe("MARK_AS_READY", nextStatus(request))),
      TE.chain(upsertSignatureRequest)
```

Reply:
If you invert the arguments in `nextStatus` you can get:

`TE.chainEitherKW(nextStatus("MARK_AS_READY"))`

for the last line

Reply:
Try to avoid nested pipelines / flow

Reply:
right, I agree!